devlib/module/gem5stats.py

```diff
@@ -107,14 +108,53 @@ def start_periodic_dump(self, delay_ns=0, period_ns=10000000):
     
     def match(self, keys, rois_labels, base_dump=0):
         '''
-        Tries to match the list of keys passed as parameter over the statistics
-        dumps covered by selected ROIs since ``base_dump``. Returns a dict 
-        indexed by key parameters containing a dict indexed by ROI labels 
-        containing an in-order list of records for the key under consideration
-        during the active intervals of the ROI.
-
-        Keys must match fields in gem5's statistics log file. Key example:
-            system.cluster0.cores0.power_model.static_power
+        Extract specific values from the statistics log file of gem5
+
+        :param keys: list of regex matched over the fields of the statistics 
+            file. match() returns only the values of fields matching at least
+            one these regex.
+        :type keys: list
+
+        :param rois_labels: list of ROIs labels. match() returns the values of
+            the specified fields only during dumps spanned by at least of
+            of these ROIs.
+        :type rois_label: list
+
+        :param base_dump: dump number from which match() should operate. By 
+            specifying a non-zero dump number, one can virtually truncate 
+            the head of the stats file and ignore all dumps before a specific
+            instant. The value of ``base_dump`` will typically (but not 
+            necessarily) be the result of a previous call to ``next_dump_no``. 
+        :type base_dump: int
+
+        :returns: a dict indexed by key parameters containing a dict indexed by
+        ROI labels containing an in-order list of records for the key under
+        consideration during the active intervals of the ROI. 
+        
+        Example of return value:
+         * Result of match(['sim_'],['roi_1']):
+            {
+                'sim_inst': 
+                {
+                    'roi_1': ['265300176', '267975881']
```

Comment:
Hmm good point. The current implementation returns only strings (or sets of strings). It looks like the values in the stats file are printed in many different formats, but most of them can be converted easily using `devlib.utils.types.numeric`. However, for the others, it might be more tricky. For example, some stats values are percentages, and they usually include the `%` symbol which will cause issues during the cast. I'm not sure if this is the only problematic case and if we can safely assume anything about the format of values dumped in the stats file.

@Sticklyman1936 , @avanlaer : are we guaranteed to have numeric values only in the stats file ? Is the `%` symbol the only corner case ?